src/client.rs

```diff
@@ -517,4 +517,10 @@ impl Default for Timeout {
 
 // pub(crate)
 
-pub struct KeepCoreThreadAlive(Arc<InnerClientHandle>);
+pub struct KeepCoreThreadAlive(Option<Arc<InnerClientHandle>>);
+
+impl Default for KeepCoreThreadAlive {
```

Comment:
`Default` implies to me that it's the normal value. Since it's abnormal to create this without a reference, I'd probably opt for changing this to an inherent `pub(crate) fn empty() -> Self`. I think it makes it just a little clearer.